pkg/command/root.go

```diff
@@ -816,13 +817,29 @@ func shouldSkipGlobalInit(cmd *cobra.Command) bool {
 	return isSkipCommand(skipGlobalInitCommands, cmd.CommandPath())
 }
 
+var globalRootCmd *cobra.Command
+var globalRootCmdLock = &sync.Mutex{}
+
+// NewRootCmd create a new root command instance if it was not created before
+func NewRootCmd() (*cobra.Command, error) {
+	globalRootCmdLock.Lock()
```

Comment:
Since the unit tests will aways first set `globalRootCmd = nil`, I don't think the locking is necessary here anymore.  But, to be really safe, we may want to add it in `NewRootCmdForTests()`

Reply:
Had a discussion and we decided to keep the looking to protect us against the goroutine sceneario mentioned by Vui.